parser/parser.go

```diff
@@ -107,7 +108,9 @@ func (p *parser) parseProgram() *ast.Program {
 			}
 		}
 
-		if stmt != nil && strings.TrimSpace(stmt.String()) != "" {
+		if stmt != nil &&
+			(reflect.ValueOf(stmt).Kind() == reflect.Ptr && !reflect.ValueOf(stmt).IsNil()) &&
```

Comment:
I would like to prevent using reflect in this loop. Too much reflection, and actually this is not necessary. I will leave another comment for the solution, based on the real issue.

Reply:
@sio4 , Yes, there's another way to update the `parseLetStatement()` function 

 ```
 if !p.expectPeek(token.ASSIGN) {
    return nil
}
``` 

to return `stmt` instead of `nil`. However, we will have to update all the parser functions to not return nil as calling `stmt.String()` will panic if any of the parser functions return `nil`. I made the decision to use reflection as it's the safest route considering how heavily this package relies on interfaces{} and reflections.

Reply:
Correct. also, we can fix `parseStatement()`, and that could be the best. (See my another comment) With your PR, the for loop on tokens will reflect all statements returned and it is expensive and not necessary.